bindings/java/src/main/java/org/ucx/jucx/ucp/UcpWorkerParams.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.ucp;
+
+import java.nio.ByteBuffer;
+import java.util.BitSet;
+
+import org.ucx.jucx.UcxParams;
+import org.ucx.jucx.ucs.UcsConstants;
+
+public class UcpWorkerParams extends UcxParams {
+
+    private int threadMode;
+
+    private BitSet cpuMask = new BitSet();
+
+    private long events;
+
+    private ByteBuffer userData;
+
+    private int eventFD;
+
+    @Override
+    public UcpWorkerParams clear() {
+        super.clear();
+        threadMode = 0;
+        cpuMask = new BitSet();
+        events = 0;
+        userData = null;
+        eventFD = 0;
+        return this;
+    }
+
+    /**
+     * Requests the thread safety mode which worker and the associated resources
+     * should be created with.
+     * When thread safety requested, the {@link UcpWorker#UcpWorker(UcpContext)}
+     * attempts to create worker where multiple threads can access concurrently.
+     * The thread mode with which worker is created can differ from the
+     * suggested mode.
+     */
+    public UcpWorkerParams requestThreadSafety() {
```

Comment:
It was at first. @evgeny-leksikov mentioned that basically there are only 2 thread modes interested in java (other's for MPI only): single (that by default), and multy (thread safety).

Reply:
i think that all thread modes are relevant for any application, though SINGLE and MULTI obviously are more frequently used. Since we want a generic API here, would be best to support all options. @evgeny-leksikov WDYT?

Reply:
Actually, I did not see any real case of using serialized mode. I thought it's just matching to MPI standard. BTW, typically MPI implements only single and multi modes as well :)